internal/execute/outputs.go

```diff
@@ -32,17 +32,16 @@ func getFormatOptsOfSys(sys System) *diagnosticwriter.FormattingOptions {
 type diagnosticReporter = func(*ast.Diagnostic)
 
 func quietDiagnosticReporter(diagnostic *ast.Diagnostic) {}
-func createDiagnosticReporter(sys System, options *core.CompilerOptions) diagnosticReporter {
+func createDiagnosticReporter(sys System, w io.Writer, options *core.CompilerOptions) diagnosticReporter {
```

Comment:
Seems suspect to take both System and a writer, but I do see why that changed here. It sort of makes me wonder if `System` should even have the writer on it.

Reply:
i think its ok as it finally goes onto `system.writer()` only , its just we create string buffer to get output and when its ready to stream to output is when we stream it to system.writer()